source/read-write-concern/read-write-concern.rst

```diff
@@ -517,6 +517,30 @@ have been abbreviated:
 - ``{ok:1, writeConcernError: {code: 100, codeName: "UnsatisfiableWriteConcern", errmsg: "Not enough data-bearing nodes"}}``
 - ``{ok:1, writeConcernError: {code: 79, codeName: "UnknownReplWriteConcern"}}``
 
+A writeConcernError may include an "errInfo" field providing additional information (e.g. the source of the write concern associated with the error, which is useful when ). Here is an example:
```

Comment:
I've now made it explicit in the CRUD spec that `details` should be set to `errInfo`.

I agree the specs should tie together the expected server errors and how drivers should represent them. But I am hesitant to move all of server error handling of read/write concern into CRUD spec since the scope of this ticket is small. I'd vote for completing this change now, since it was explicitly noted as a downstream change of the addition of the "provenance" field in 4.4 ([SERVER-45623](https://jira.mongodb.org/browse/SERVER-45623)), and creating/editing tickets for proposed improvements to error handling, and how it is tested. Some other thoughts:

- There are helpers in the read/write concern spec that aren't in CRUD (e.g. `createUser`). If I move the error handling notes there, then the read/write concern spec would still rely on CRUD for complete information on what to do with write concern errors.
- There are open SPEC tickets to indicating the CRUD spec isn't quite right to begin with. Moving the error handling there may require addressing those as well:
    - https://jira.mongodb.org/browse/SPEC-1199 (How should drivers handle multiple write concern errors)
    - https://jira.mongodb.org/browse/SPEC-1492 (Crud spec claims only one of writeError/writeConcernError will be populated at a time)
- If we move all of error handling to the CRUD spec, would we do the same with the bulk write error handling logic? That may be better left to when the bulk write spec is rewritten.
- In addition to moving the error handling specification closer to the error/exception types defined, I think we should have better spec tests around errors that CRUD methods can throw. Perhaps instead of just specifying `errorContains`, `errorCodeName`, use the same error types we define.

Reply:
That all makes sense. I think the changes you've made address most of the concerns I mentioned above. 

Once this PR gets merged, we can file a "Consolidate write concern error handling" ticket that we can get done at some point in the future, perhaps as part of writing the new bulk write spec.

Reply:
That sounds good to me, I'll file after this is approved.

Reply:
https://jira.mongodb.org/browse/DRIVERS-965